ActionSheet.android.js

```diff
@@ -110,26 +124,74 @@ class ActionGroup extends React.Component {
           onPress={() => onSelect(i)}
           style={styles.button}>
           {iconElement}
-          <Text style={[styles.text, { color }, textStyle]}>{options[i]}</Text>
+          <Text style={[styles.text, textStyle, { color }]}>{options[i]}</Text>
         </TouchableNativeFeedbackSafe>
       );
 
-      if (i < startIndex + length - 1) {
-        optionViews.push(<View key={`separator-${i}`} style={styles.rowSeparator} />);
+      if (showSeparators && i < startIndex + length - 1) {
+        optionViews.push(this._renderRowSeparator(i));
       }
     }
 
     return (
       <View style={styles.groupContainer}>
+        {this._renderTitleContent()}
         <ScrollView>{optionViews}</ScrollView>
       </View>
     );
   }
+
+  _renderRowSeparator(key) {
+    return <View key={key ? `separator-${key}` : null} style={styles.rowSeparator} />;
+  }
+
+  _renderTitleContent()  {
+    const { title, titleTextStyle, message, messageTextStyle, showSeparators } = this.props;
+
+    if (!title && !message) {
+      return null;
+    }
+
+    return (
+      <View>
+        <View style={[styles.titleContainer, { paddingBottom: showSeparators ? 24 : 16 }]}>
+          {title ? <Text style={[styles.title, titleTextStyle]}>{title}</Text> : null}
+          {message ? <Text style={[styles.message, messageTextStyle]}>{message}</Text> : null}
+        </View>
+        {showSeparators ? this._renderRowSeparator('title') : null}
```

Comment:
you can do `showSeparators && this._renderRow ...` here

Reply:
as long as showSeparators will never === 0, this works.

Reply:
leave this or update to something like `!!showSeparators && this._renderRow ...`?

Reply:
so long as showSeparators is always a boolean (or else it'll crash android) you can do ` && this._renderRow` - even `Boolean(showSeparators) && this._renderRow` would work.

Reply:
showSeparators is an optional boolean prop, so it wouldn't actually work as suggested. So i'll do the !! in front to cast the boolean or undefined to boolean.